include/extractor/class_data.hpp

```diff
@@ -0,0 +1,20 @@
+#ifndef OSRM_EXTRACTOR_CLASSES_DATA_HPP
+#define OSRM_EXTRACTOR_CLASSES_DATA_HPP
+
+#include "util/bit_range.hpp"
+
+#include <cstdint>
+
+namespace osrm
+{
+namespace extractor
+{
+
+using ClassData = std::uint8_t;
+static const std::uint8_t MAX_CLASS_INDEX = 8 - 1;
```

Comment:
If we constraint it to 8 bits to keep the class data small we should tell users this in the changelog/docs/profile docs.

(Btw overflowing this by setting a higher number from Lua should just wrap around and have the usual unsigned behavior, no? Should we check that in our Lua-exposed setter function?)